pkg/infrastructure/services/messaging_kafka.go

```diff
@@ -58,7 +58,7 @@ func (k *kafkaMessagingDeployer) createRequiredKafkaTopics(infra *v1alpha1.Kogit
 		}
 	}
 	// topics required by the deployed service
-	topics, err := k.fetchRequiredTopics(service)
+	topics, err := k.fetchTopicsAndSetCloudEventsStatus(service)
```

Comment:
In `fetchTopicsAndSetCloudEventsStatus` function we are fetching KafkaTopics by making `Get` request to service where in `createKafkaTopicIfNotExists` function we are fetching topics from `ServiceDefinition` object `k.definition.KafkaTopics`. Can we replace `k.definition.KafkaTopics` to fetch kafka topic list by making `Get` call?

Reply:
> Can we replace k.definition.KafkaTopics to fetch kafka topic list by making Get call?

We can't right do it right now since those topics are not been defined in the topic discovery since it's specific to each supporting service.

What we should do is implement this discovery on Data Index, Management Console and so on, so we could remove it. I'll open a JIRA to track this.

Reply:
JIRA: https://issues.redhat.com/browse/KOGITO-3792

Thanks for bringing this up, completely slipped from my mind.